lib/models.js

```diff
@@ -603,11 +679,31 @@ class Cardinality {
   toString() {
     return `${this.min}..${this.isMaxUnbounded ? '*' : this.max}`;
   }
+
+  toJSON() {
+    var out = {
+      "min": this._min,
+      "max": this._max
+    };
+
+    if (this._history && this._history.length > 0) {
+      out["history"] = this._history.map(h => ({
+        "source": h["source"],
+        "min": h._min,
+        "max": h._max
+      }))
+    }
+
+    return out;
+
+  }
 }
 
 class Constraint {
+ 
   constructor(path = []) {
     this._path = path;
+    this._lastModifiedBy = undefined;
```

Comment:
to copy from elsewhere: "an undefined field will not be translated to JSON; however, a null or otherwise empty field will output to "baz": "", which resulted in unnecessary verbosity. Especially as key existence was intended to be a reliable method of identifying if a property exists."